nano/node/confirmation_height_unbounded.cpp

```diff
@@ -469,6 +481,15 @@ bool nano::confirmation_height_unbounded::has_iterated_over_block (nano::block_h
 	return block_cache.count (hash_a) == 1;
 }
 
+bool nano::confirmation_height_unbounded::has_iterated_or_confirmed (nano::block_hash const & hash_a) const
```

Comment:
Can you add a comment to explain what this function does?
I do not immediately obvious what iterated means in this context.
Also, it would be nice to make it clear if confirmed means cemented in this case.

Reply:
There is an existing function called `has_iterated_over_block` .
Here `has_iterated` means the same.
A straightforward implementation would involve calling the existing `has_iterated_over_block` and a new `is_recently_confirmed` each acquiring and releasing the mutex lock independently.
Something like : `return has_iterated_over_block() || is_recently_confirmed()`

 I opted to implement `has_iterated_or_confirmed` by duplicating the logic from both methods under a single mutex lock acquisition.